node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/CuratorDatabaseClient.java

```diff
@@ -84,6 +84,8 @@ private void initZK() {
         for (Node node : nodes) {
             if (node.state() != expectedState)
                 throw new IllegalArgumentException(node + " is not in the " + node.state() + " state");
+
+            node = node.with(node.history().recordStateTransition(null, expectedState, Agent.system, clock.instant()));
```

Comment:
So, did you figure out what happens if expectedState is reserved here? Add comment?

Reply:
Didn't really look, I don't think it should matter either way if the node immediately goes from `reserved` straight to `active`?